# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1524601760 25200
#      Tue Apr 24 13:29:20 2018 -0700
# Branch stable
# Node ID a6419af900314b3728c94913dc0161afe4727489
# Parent  47c803b4b8aee6b56e4a4a853eee6630237fa1e2
pkg: restore deletion of package archive downloaded from Octave Forge

* pkg.m: Restore code to delete all downloaded files, whether they were
downloaded to a temporary directory or not.  Fixes regression introduced
in cset 0cd7076a97c8.

diff --git a/scripts/pkg/pkg.m b/scripts/pkg/pkg.m
--- a/scripts/pkg/pkg.m
+++ b/scripts/pkg/pkg.m
@@ -433,16 +433,17 @@ function [local_packages, global_package
 
             endfor
           endif
         endif
         install (files, deps, prefix, archprefix, verbose, local_list,
                  global_list, global_install);
 
       unwind_protect_cleanup
+        cellfun ("unlink", local_files);
         if (exist (tmp_dir, "file"))
           rmdir (tmp_dir, "s");
         endif
       end_unwind_protect
 
     case "uninstall"
       if (isempty (files))
         error ("pkg: uninstall action requires at least one package name");
